packages/elements/src/time-picker/index.ts

```diff
@@ -722,28 +732,23 @@ export class TimePicker extends ControlElement {
   /**
    * Changes a time segment value by a specified amount.
    * Also updates parent values when rolling through cycles.
+   * Incomplete value will update only segment without pre-populate value.
    * @param amount Amount to change by
    * @param segment Segment id
    * @returns {void}
    */
   private changeValueBy(amount: number, segment: Segment): void {
-    let offset = 0;
-    switch (segment) {
-      case Segment.HOURS:
-        offset = this.hours === null ? 0 : amount * MILLISECONDS_IN_HOUR;
-        break;
-      case Segment.MINUTES:
-        offset = this.minutes === null ? 0 : amount * MILLISECONDS_IN_MINUTE;
-        break;
-      case Segment.SECONDS:
-        offset = this.seconds === null ? 0 : amount * MILLISECONDS_IN_SECOND;
-        break;
-      // no default
+    const segmentValue = this[segment];
+    const { miliseconds, cycle } = SegmentMap[segment];
+
+    if (this.isCompleteValue) {
+      const offset = segmentValue === null ? 0 : amount * miliseconds;
+      const value = addOffset(this.currentTimeString, offset);
+      this.setValueAndNotify(value);
+      this.selectedSegment = segment;
+    } else {
+      this[segment] = segmentValue === null ? 0 : (segmentValue + amount + cycle) % cycle;
```

Comment:
why we need to `cycle` in `(segmentValue + amount + cycle)`? It looks redundant.

Reply:
To make it can cycling through to max value. 
For example 0 back to 23 

segment = 'hours'
segmentValue = 0
amount = -1
cycle = 24

this['hours'] = (segmentValue + amount + cycle) % cycle
this['hours'] = (0 + (-1) + 24) % 24
this['hours'] = 23 % 24
this['hours'] = 23

Reply:
Negative amount! Thx.